mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3746,6 +3748,31 @@ int main(int argc, char **argv) {
   llvm::cl::ParseCommandLineOptions(argc, argv,
                                     "MLIR Rock Dialect host generation\n");
 
+  if (!arch.getValue().empty()) {
```

Comment:
Important for fixing the bug -- the typeFromString/strToType changes aren't enough, because the latter runs second and isn't aware of architecture.